Tests/CursorTests.swift

```diff
@@ -0,0 +1,194 @@
+            XCTAssertEqual(cursor?.position, 42)
+
+            ex.fulfill()
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testNewReadCursorHook() {
+        let ex = expectation(description: "received new read cursor")
+
+        let aliceRoomDelegate = AliceRoomDelegate(expectation: ex)
+        alice!.subscribeToRoom(
+            room: alice!.rooms.first(where: { $0.id == roomId! })!,
+            roomDelegate: aliceRoomDelegate
+        )
+
+        sleep(1)
+
+        bob?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testGetAnotherUsersReadCursorBeforeSubscribingFails() {
+        let ex = expectation(description: "got another users read cursor fails")
+
+        bob?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+
+            do {
+                let _ = try alice!.readCursor(roomId: roomId!, userId: "bob")
+            } catch let error {
+                switch error {
+                case PCCurrentUserError.noSubscriptionToRoom:
+                    ex.fulfill()
+                default:
+                    XCTFail()
+                }
+            }
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testGetAnotherUsersReadCursor() {
+        let ex = expectation(description: "got another users read cursor fails")
+
+        let aliceRoomDelegate = AliceRoomDelegate()
+        alice!.subscribeToRoom(
+            room: alice!.rooms.first(where: { $0.id == roomId! })!,
+            roomDelegate: aliceRoomDelegate
+        )
+
+        sleep(1)
+
+        bob?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+
+            sleep(1) // give the read cursor a chance to propagate down the connection
+            let cursor = try! alice!.readCursor(roomId: roomId!, userId: "bob")
+            XCTAssertEqual(cursor?.position, 42)
+
+            ex.fulfill()
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func user(id: String, delegate: PCChatManagerDelegate = TestingChatManagerDelegate()) -> PCCurrentUser {
+        var user: PCCurrentUser?
+
+        let ex = expectation(description: "connected as user with ID \(id)")
+
+        let chatManager = ChatManager(
+            instanceLocator: testInstanceLocator,
+            tokenProvider: PCTokenProvider(url: testInstanceTokenProviderURL),
+            userId: id,
+            logger: TestLogger()
+        )
+
+        chatManager.connect(delegate: delegate) { u, error in
+            XCTAssertNil(error)
+            XCTAssertNotNil(u)
+
+            user = u
+            ex.fulfill()
+        }
+
+        waitForExpectations(timeout: 5)
+        return user!
+    }
+
+    func createRoom(user: PCCurrentUser, roomName: String, addUserIds: [String] = []) -> PCRoom {
+        var room: PCRoom?
```

Comment:
This is just a comment on usage of Swift more than anything but you could define room as `PCRoom!` here and then when you return you would just `return room` (i.e. no need for the `room!` in the return)

Reply:
noooooice. This is exactly what I actually wanted.